cloudinit/subp.py

```diff
@@ -286,27 +250,15 @@ def subp(
             shell=shell,
             cwd=cwd,
         )
-        (out, err) = sp.communicate(data)
+        out, err = sp.communicate(data)
     except OSError as e:
-        if status_cb:
-            status_cb("ERROR: End run command: invalid command provided\n")
         raise ProcessExecutionError(
             cmd=args,
             reason=e,
             errno=e.errno,
             stdout="-" if decode else b"-",
             stderr="-" if decode else b"-",
         ) from e
-    finally:
-        if devnull_fp:
-            devnull_fp.close()
-
-    # Just ensure blank instead of none.
-    if capture or combine_capture:
```

Comment:
@TheRealFalcon I suppose it would probably be better to include a note in the commit message summarizing this (or making it its own separate commit) so that this doesn't look like a mistake in the git log.